Adicionado modularizar.py para modularizar o conteudo

BrantaRodiney opened the pull request:
O conteudo esta no formato funcional, e preciso modularizar seguindo os seguitnes parametros:

1.	Coloque o código que você fez nas etapas 1 e 2 dentro de uma única função chamada “obter_limite”. Essa função deverá retornar o limite que o usuário poderá gastar.

2.	Coloque o código que você fez na etapa 3 dentro de uma única função chamada “verificar_produto”. Essa função terá como parâmetro de entrada o limite de gasto do cliente.
Após o cliente informar os dados dele (pela função “obter_limite”), armazene o limite que ele poderá gastar dentro de uma variável chamada “limite”.

3.	Na sequência, pergunte ao usuário quantos produtos deseja cadastrar.
Por fim, utilize uma estrutura de repetição (for ou while) por n vezes, com n equivalendo à quantidade de produtos que ele deseja cadastrar.
Dentro da sua estrutura de repetição, chame sua função “verificar_produto”, permitindo que o usuário consiga cadastrar todos os produtos e verificar se teria ou não limite sobrando para comprá-los.

BrantaRodiney committed:
Adicionado modularizar.py para converter de codigo funcional para modular

marcocarvalho commented on main.py:
```diff
@@ -1,15 +1,45 @@
+#     print(f"\nSeu cargo atual: {cargo_atual} \nSeu Salario atual: {salario_atual} \nSeu ano de nascimento: {ano_de_nascimento}")
+#     print(f"Voce tem aproximadamente {idade_aproximada} anos.")
+#     print(f"Voce possui o valor em credito de R$ {valor_credito_disponivel}, disponiveis para compras em nossa loja.")
+#     return valor_credito_disponivel
```
Não se commita código comentado
```suggestion
```

marcocarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,62 @@
+
+print("", ContaCaracter(meuNomeCompleto))
+print("", CaracteresPrimeiroNome(meuNomeCompleto))
+print("", LiberaDesconto(ValorProduto, ValorProduto, meuNomeCompleto))
```
linhas, sempre terminam com um enter.

Configure seu editor para adicionar um enter no final de cada linha.

Qual editor está usando?

BrantaRodiney replied:
Visual Studio Code - Version: 1.44.2
Configured the editor and updated the files.

marcocarvalho replied:
![image](https://user-images.githubusercontent.com/503670/80897112-de0f0f00-8ccb-11ea-9392-27fed9ce9182.png)

BrantaRodiney commented on main.py:
```diff
@@ -1,15 +1,45 @@
-    ano_vigente = datetime.date.today()
-    idade_aproximada = (ano_vigente.year - ano_de_nascimento)
+
+NOME_COMPLETO = "Rodiney Branta"
```
Its a constant

marcocarvalho commented on main.py:
```diff
@@ -1,15 +1,45 @@
-    print(f"Seu cargo atual: {cargo_atual} \nSeu Salario atual: {salario_atual} \nSeu ano de nascimento: {ano_de_nascimento}")
-    print(f"Voce tem aproximadamente {idade_aproximada} anos.")
-    print(f"Voce possui o valor em credito de R$ {valor_credito_disponivel:.2f}, disponiveis para compras em nossa loja.")
+    return valor_credito_disponivel
```
Eu retiraria a variável, não tem pq ela existir

```suggestion
    return ((salario_atual*(idade_aproximada / 1000)) +100) 
```
e removeria a linha de cima, claro.

adilsoncarvalho replied:
Olhando o fonte, ficou estranho, pois o nome da função é `obter_limite`, mas a variável com o valor a ser retornado se chama `valor_credito_disponivel`. Meio confuso isso. Estamos aí calculando o que?

BrantaRodiney replied:
> Olhando o fonte, ficou estranho, pois o nome da função é `obter_limite`, mas a variável com o valor a ser retornado se chama `valor_credito_disponivel`. Meio confuso isso. Estamos aí calculando o que?

O objetivo da função é retornar o limite de gastos.

marcocarvalho commented on main.py:
```diff
@@ -1,15 +1,45 @@
+ano_vigente = datetime.date.today()
+idade_aproximada = (ano_vigente.year - ano_de_nascimento)
+
+os.system("cls" if os.name == "nt" else "clear")
```
Procure na stdlib do python como limpar a tela do console. Isso é gambs.

adilsoncarvalho replied:
https://www.google.com/search?q=how+to+clear+the+console+in+python&oq=how+to+clear+the+console+in+p&aqs=chrome.0.0j69i64j5l6.8247j0j4&sourceid=chrome&ie=UTF-8

BrantaRodiney replied:
Removido o modulo e linha, que detecta o sistema operacional, e limpa a tela de maneira não profissional.

BrantaRodiney replied:
> Procure na stdlib do python como limpar a tela do console. Isso é gambs.

[Modulo Curse](https://docs.python.org/3.8/library/curses.html)
Não está incluso no pacote Python para Windows, mas pode ser adicionado.
Este método pode ser considerado uma forma mais profissional de controle de telas no modo texto?

BrantaRodiney replied:
> Procure na stdlib do python como limpar a tela do console. Isso é gambs.

O Python não possui nenhuma função (build-in) para tratar de limpeza de console.
Para escape sequence via: print("\033[1;1f").  
Mas é visto como falha de segurança, e somente vai funcionar em terminais que aceitem isto.

marcocarvalho commented on main.py:
```diff
@@ -1,15 +1,45 @@
+print(f"Voce tem aproximadamente {idade_aproximada} anos.")
+print(f"Voce possui o valor em credito de R$ {limite:.2f}, disponiveis para compras em nossa loja.")    
+
+time.sleep(5)
```
pq o sleep? Pq deliberadamente vc está deixando o seu código mais lento?

BrantaRodiney replied:
Fiz o uso do sleep para que as informações fiquem na tela, caso o programa seja executado fora do console, e possibilitem a leitura antes do encerramento da execução.

marcocarvalho replied:
Não seria melhor um "aperte qualquer tecla para sair" ?

BrantaRodiney replied:
```suggestion
print("\nAperte qualquer tecla para sair")
input()
```

adilsoncarvalho replied:
Exato, o @marcocarvalho tá certo. Aqui é o glorioso momento de um "Pressione qualquer tecla para continuar.", e piadas de "no meu teclado não tem a tecla _qualquer tecla_" já perderam a graça desde o Windows 95 🤣

adilsoncarvalho replied:
> 

@BrantaRodiney aqui vc como o autor do PR não _sugere alteração_, vc vai lá e altera. Quando fizer todas as alterações, a gente faz um novo review.

BrantaRodiney replied:
> > 
> 
> @BrantaRodiney aqui vc como o autor do PR não _sugere alteração_, vc vai lá e altera. Quando fizer todas as alterações, a gente faz um novo review.

De acordo!

BrantaRodiney replied:
> Não seria melhor um "aperte qualquer tecla para sair" ?

Adicionado função print do tipo string, informando ao usuário que poderá sair do programa se apertar o botão ENTER

marcocarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,62 @@
+
+print("Seja bem-vindo a loja de coelhos, entre e escolha um Rodiney Branta")
+
+Cargo_Atual = str(input("Para fazermos sua analise de credito, precisamos que voce entre com algumas informacoes:\nCargo atual:"))
+Salario_Atual = float(input("Salario Atual:"))
+Ano_Nascimento = int(input("Ano de nascimento:"))
```
Se o objetivo é modularizar o código, pq vc está repetindo código aqui?

marcocarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,62 @@
+Credito_Atual = ((Salario_Atual*(Idade_Aproximada / 1000)) +100) #ATP_ETAPA2
+meuNomeCompleto = "Rodiney Branta"
+
+os.system("cls")
```
gambs... ✂️

marcocarvalho replied:
https://pypi.org/project/clear-screen/

marcocarvalho replied:
ou melhor ainda.. entender O QUE É LIMPAR A TELA: https://www.quora.com/Is-there-a-Clear-screen-function-in-Python/answer/Jim-Dennis-1

BrantaRodiney replied:
> https://pypi.org/project/clear-screen/

[Commit](https://github.com/BrantaRodiney/CompThinking/pull/1/commits/b77fc925b28aec59d64fc2093cd049aafd17afc3) efetuado para correção através de uma função que limpa a tela, escrita por mim, já que o Python, não possui uma função built-in para tratar este problema.
O terminal que utilizo aceita character escaping através de um _print("\033[1;1f")_ – [ansi-escape sequence vt-100](http://ascii-table.com/ansi-escape-sequences-vt-100.php), entretanto observei que devido ao caracteres especiais, há casos, que o usuário pode utilizar deste recurso para ver algo que não deveria, e o programa acha que esta  executando outras coisas.

marcocarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,62 @@
+
+
+def ContaCaracter(contador):
+    resultado =  len(contador)
+    return resultado
```
```suggestion
    return len(contador)
```

marcocarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,62 @@
+def CaracteresPrimeiroNome(meuNomeCompleto):
+    primeiroNome = meuNomeCompleto.find(' ')
+    if primeiroNome != -1:
+        return(primeiroNome)
```
se 

"Marco Carvalho".find(' ') == 5... ok... mas se o python a primeira posição de uma string é zero.. seu  código vai retornar 4 pra "Marco Carvalho"

BrantaRodiney replied:
Seguindo a condição de ter 2 funções e o nome do representante da loja ser fixo, utilizei a função len() para fazer a contagem da constante definida no código.

marcocarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,62 @@
+    return resultado
+
+
+def CaracteresPrimeiroNome(meuNomeCompleto):
```
Tem uma forma menos  if/else de resolver isso daqui...

BrantaRodiney replied:
> Tem uma forma menos if/else de resolver isso daqui...

Efetuado alteração que fez split do nome para uma lista e utilizado função len() para fazer a contagem de caracteres

BrantaRodiney committed:
Removido o codigo comentado

BrantaRodiney committed:
Insert final new line configuration added
configured the editor to insert final newline at the end of the file
when saving it.

BrantaRodiney committed:
Replaced the variable to return only

Co-authored-by: Marco Carvalho <marco.carvalho.swasthya@gmail.com>

adilsoncarvalho commented on main.py:
```diff
@@ -1,15 +1,29 @@
+print(f"Seja bem-vindo a Poppy loja , sou {NOME_COMPLETO}, entre e escolha um produto")
+
+cargo_atual = str(input("Para fazermos sua analise de credito, "
+                        "solicitamos que voce entre com algumas informacoes:\nCargo atual: "))
```
Sempre cuide para isolar as responsabilidades no código fonte. Há momentos em que parece preciosismo, mas acredite que no _long run_, a gente sempre sai ganhando com isto.

Aqui eu não vejo porque fixar a mensagem de boas-vindas com o pedido do cargo da pessoa. Se você precisar em algum momento adicionar uma pergunta antes, vai ter de mover a mensagem de boas-vindas para esta nova pergunta, ou pode ocorrer de esquecer e ela aparecer onde não deveria.

```suggestion
print(f"Para fazermos sua análise de crédito, solicitamos que você entre com algumas informações:")
cargo_atual = str(input("Cargo atual: "))
```

BrantaRodiney replied:
> Sempre cuide para isolar as responsabilidades no código fonte. Há momentos em que parece preciosismo, mas acredite que no _long run_, a gente sempre sai ganhando com isto.
> 
> Aqui eu não vejo porque fixar a mensagem de boas-vindas com o pedido do cargo da pessoa. Se você precisar em algum momento adicionar uma pergunta antes, vai ter de mover a mensagem de boas-vindas para esta nova pergunta, ou pode ocorrer de esquecer e ela aparecer onde não deveria.

Estou estudando uma forma, de mover a mensagem de boas-vindas e modularizar o while, para que eu possa utilizar `if __name__ == '__função__': ` para controlar a chamada das funções do modulo modularizando.py.

adilsoncarvalho commented on main.py:
```diff
@@ -1,15 +1,29 @@
+                        "solicitamos que voce entre com algumas informacoes:\nCargo atual: "))
+salario_atual = float(input("Salario Atual: "))
+ano_de_nascimento = int(input("Ano de nascimento: "))
+ano_vigente = datetime.date.today()
```
```
ano_vigente = datetime.date.today()
```

Isto aqui eu moveria para logo abaixo da mensagem de boas-vindas. Não é uma regra, mas é sempre bom deixar todas as variáveis auto atribuídas já no início do bloco da qual fazem parte. Com isto, fica mais simples de achar o que você atribuiu das que você solicitou para o usuário preencher.

BrantaRodiney replied:
> ```
> ano_vigente = datetime.date.today()
> ```
> 
> Isto aqui eu moveria para logo abaixo da mensagem de boas-vindas. Não é uma regra, mas é sempre bom deixar todas as variáveis auto atribuídas já no início do bloco da qual fazem parte. Com isto, fica mais simples de achar o que você atribuiu das que você solicitou para o usuário preencher.

Atualizado em modularizar.py e removido de main.py

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,61 @@
+
+print("Seja bem-vindo a loja de coelhos, entre e escolha um Rodiney Branta")
+
+Cargo_Atual = str(input("Para fazermos sua analise de credito, precisamos que voce entre com algumas informacoes:\nCargo atual:"))
```
Eu não codo em python, mas este padrão de nome de variável me parece bizarro. Geralmente a gente usa ou `camel case` ou `snake case`, mas nunca ambos ao mesmo tempo.

Este [post no Medium](https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841) explica os _cases_ existentes de uma forma bem bacana. É bom você consultar o guida do python para padrões de código.

São vários, mas a regra de ouro é: _em Roma, faça como os romanos_.

> Python and Ruby both recommend UpperCamelCase for class names, CAPITALIZED_WITH_UNDERSCORES for constants, and lowercase_separated_by_underscores for other names.

> In Python, if a name is intended to be "private", it is prefixed by an underscore. Private variables are enforced in Python only by convention. Names can also be suffixed with an underscore to prevent conflict with Python keywords. Prefixing with double underscores changes behaviour in classes with regard to name mangling. Prefixing and suffixing with double underscores are reserved for "magic names" which fulfil special behaviour in Python objects.

https://en.wikipedia.org/wiki/Naming_convention_(programming)#Python_and_Ruby

Eu acho que isto aqui poderia ser outro PR, um que se preocupa exclusivamente com arrumar os nomes das funções e variáveis.

BrantaRodiney replied:
Seguindo o [PEP 8 -- Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions) e o link que você indicou, atualizei os nomes de variáveis, funções e variáveis.

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,61 @@
+
+os.system("cls")
+
+print(f"Seu cargo atual: {Cargo_Atual} \nSeu Salario atual: {Salario_Atual} \nSeu ano de nascimento: {Ano_Nascimento}")
```
Eu não vejo o que Maria leva nesta concatenação em uma string só. Se é para fins acadêmicos, então é perfeitamente aceitável, mas no dia-a-dia da programação, ia levar na mão com uma palmatória 🤣.

O motivo é o mesmo de meu comentário anterior: isole as responsabilidades sempre. 

```suggestion
print(f"Seu cargo atual: {Cargo_Atual}")
print(f"Seu Salario atual: {Salario_Atual}")
print(f"Seu ano de nascimento: {Ano_Nascimento}")
```

BrantaRodiney replied:
> print(f"Seu cargo atual: {Cargo_Atual}")
> print(f"Seu Salario atual: {Salario_Atual}")
> print(f"Seu ano de nascimento: {Ano_Nascimento}")

Implementadas as sugestões, a princípio, utilizei para verificar como funcionava o `print()` com as quebras de linha.

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,61 @@
+NomeProduto=str(input("Informe o nome de um produto que voce deseja: "))
+ValorProduto=float(input("Informe o valor deste produto: "))
+
+# print("Produto: {} \nValor: R$ {}".format(NomeProduto, ValorProduto))
```
Sempre que puder, evite commitar comentários de código. Isso dá a entender que ele não é relevante o suficiente para fazer parte do fonte, mas não é irrelevante o suficiente para ser apagado.

BrantaRodiney replied:
Removido todos os comentários.

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,61 @@
+
+# print("Produto: {} \nValor: R$ {}".format(NomeProduto, ValorProduto))
+
+avaliar = 100 * ValorProduto / Credito_Atual
+
+ #Avaliar quando esta liberado ou bloqueado ao parcelar
+if avaliar <= 60:
+     print("LIBERADO")
+elif avaliar <= 90:
+    print("Liberado a parcelar em até 2 vezes")
+elif avaliar <= 100:
+    print("Liberado a parcelar em 3 ou mais vezes")
+else:
+    print("Bloqueado")
```
Aqui tem duas responsabilidades ocorrendo ao mesmo tempo. Você efetua um cálculo para determinar o _credit score_ e logo na sequência você usa isto para definir parcelas. Que tal simplificar usando somente o número de parcelas? Afinal podemos assumir que:

* `0x` - não pode parcelar, então o cadastro foi rejeitado
* `>1x` - pode _parcelar ao menos em uma vez_, então é cadastro aceito
* `>nx` - pode parcelar em n parcelas, também é cadastro aceito

BrantaRodiney replied:
> Aqui tem duas responsabilidades ocorrendo ao mesmo tempo. Você efetua um cálculo para determinar o _credit score_ e logo na sequência você usa isto para definir parcelas. Que tal simplificar usando somente o número de parcelas? Afinal podemos assumir que:
> 
> * `0x` - não pode parcelar, então o cadastro foi rejeitado
> * `>1x` - pode _parcelar ao menos em uma vez_, então é cadastro aceito
> * `>nx` - pode parcelar em n parcelas, também é cadastro aceito

Neste caso em especifico eu utilizei os parâmetros fornecidos pela universidade, mas posso fazer alterações no próximo PR

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,61 @@
+        return(len(meuNomeCompleto))
+
+
+def LiberaDesconto(simNao, valorProduto, meuNomeCompleto):
```
`simNao` é um nome que não ajuda na compreensão do código fonte. Tente sempre manter o teu código o mais legível possível, pois pode ser você daqui a uns meses a ter de dar manutenção nele 🤣

marcocarvalho replied:
uhahuahuahua essa regra é a de ouro... aquilo que  hj é "óbvio" amanhã vc já não vai lembrar mais :)

adilsoncarvalho replied:
Lembrar do que? :trollface:

BrantaRodiney replied:
Verifiquei inconsistência no código e alterei a forma que ele atua, entretanto ainda ficou um pouco “esquecível”.

adilsoncarvalho reviewed (commented):
Bacana a evolução do código, e no começo é assim mesmo. Já que o intento deste PR é arrumar melhor o fonte, eu daria mais atenção no `modularizar.py`. Ele deveria ter somente funções que você então iria consumir no `main.py`.

O teu arquivo `main.py` deve ter o script que coordena o fluxo geral, e o `modularizar.py` deve ter somente as funções e classes de apoio.

Um ponto que talvez você queira ver para outro PR é dar uma geral nos nomes de tudo e deixar no padrão do python. É duro Ler_Fonte queFica a_cada_momento TrocandoDePadrão.

BrantaRodiney committed:
Função limpar_tela e aguardar para finalizar exec

Adicionada função limpar_tela e solicita que usuario pressione qualquer
tecla para encerrar o programa apos ter sido executado por completo.

BrantaRodiney committed:
Update main.py

sugestão aceita para remover variável de função, que pode retornar o calculo de forma direta

Co-authored-by: Marco Carvalho <marco.carvalho.swasthya@gmail.com>

BrantaRodiney committed:
cadastro prod e func obterlimite, verificarproduto

Código transformado em 2 funções, estas chamadas obter_limite e
verificar_produto. A primeira verifica os dados basicos iniciais do
cliente, limite de gastos do cliente; a segunda função valida se o
produto pode ser comprado, se pode ser aplicado desconto, mostrando o
resultado com desconto na tela.
Utilizado o comando while para verificar se o cliente deseja cadastrar
diversos itens a serem comprados e seus respectivos valores, sendo estes
 calculados através das funções descritas acima.

BrantaRodiney committed:
Merge branch 'modularizando_o_codigo' of github.com:BrantaRodiney/CompThinking into modularizando_o_codigo

BrantaRodiney committed:
func obter_limit corrigida

BrantaRodiney commented:
> Bacana a evolução do código, e no começo é assim mesmo. Já que o intento deste PR é arrumar melhor o fonte, eu daria mais atenção no `modularizar.py`. Ele deveria ter somente funções que você então iria consumir no `main.py`.
> 
> O teu arquivo `main.py` deve ter o script que coordena o fluxo geral, e o `modularizar.py` deve ter somente as funções e classes de apoio.
> 
> Um ponto que talvez você queira ver para outro PR é dar uma geral nos nomes de tudo e deixar no padrão do python. É duro Ler_Fonte queFica a_cada_momento TrocandoDePadrão.

Atualizado todo código para somente duas funções e adicionado uma estrutura de repetição while, para permitir múltiplos cadastros de produtos e valores destes produtos.

Padronizado os nomes de variáveis e funções, também, foi alterado alguns dos nomes das variáveis e funções,  para que tenha uma melhor definição de sentido.

BrantaRodiney committed:
modularizar aprimorado e main controla fluxo geral

O arquivo modularizar.py teve linhas referentes a parte estática da tela
principal removidas e adicionado ao main.py.

main.py passa a  ter o script para controlar o fluxo de execução

BrantaRodiney committed:
print() unico desmembrado

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,45 @@
+        print(f"O valor do produto com descosto é: R$ {valor_total_com_desconto}")
+
+
+limite = obter_limite()
+
+conta_produtos_cadastros = int(input("\nInforme quantos produtos você deseja cadastrar: "))
+for contador in range(conta_produtos_cadastros):
+    verificar_produto(limite)
```
Este trecho deveria estar em uma função também, e ela ser invocada lá no teu arquivo `main.py`, senão não rolou a tal _modularização_.

```suggestion
def solicitar_emprestimo(nome_solicitante)
    limite = obter_limite(nome_solicitante)
    conta_produtos_cadastros = int(input("\nInforme quantos produtos você deseja cadastrar: "))
    for contador in range(conta_produtos_cadastros):
        verificar_produto(limite)
```

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,45 @@
+import datetime
+
+
+def obter_limite():
+    NOME_COMPLETO = "Rodiney Branta"
+    print(f"Seja bem-vindo a Poppy loja , sou {NOME_COMPLETO}, entre e escolha um produto\n")
```
```suggestion
def obter_limite(nome_solicitante):
    print(f"Seja bem-vindo a Poppy loja , sou {nome_solicitante}, entre e escolha um produto\n")
```

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,45 @@
+    NOME_COMPLETO = "Rodiney Branta"
+    print(f"Seja bem-vindo a Poppy loja , sou {NOME_COMPLETO}, entre e escolha um produto\n")
+    ano_atual = datetime.datetime.today()
+    cargo_atual = str(input("Para fazermos sua análise de crédito, precisamos que você entre com algumas informações:\nCargo atual:"))
```
Não misture as coisas. Como havia comentado antes, se deixar os textos misturados, para dar manutenção depois fica pior. Dar boas vindas é uma coisa, solicitar o cargo atual é outra. Veja o [single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) do [S.O.L.I.D.](https://en.wikipedia.org/wiki/SOLID)
 
```suggestion
    print(f"Para fazermos sua análise de crédito, precisamos que você entre com algumas informações:")
    cargo_atual = str(input("Cargo atual:"))
```

BrantaRodiney replied:
> Não misture as coisas. Como havia comentado antes, se deixar os textos misturados, para dar manutenção depois fica pior. Dar boas vindas é uma coisa, solicitar o cargo atual é outra. Veja o [single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) do [S.O.L.I.D.](https://en.wikipedia.org/wiki/SOLID)

Aceito a sugestão, fiz a leitura do  single-responsibility principle e vou seguir as regras do S.O.L.I.D. em meus códigos.

adilsoncarvalho commented on modularizar.py:
```diff
@@ -0,0 +1,45 @@
+    print(f"Seu Salario atual: {salario_atual}")
+    print(f"Seu ano de nascimento: {ano_de_nascimento}")
+    print(f"Voce possui R$ {credito:.2f} em créditos disponíveis para compras em nossa loja.")
+    return credito, idade_aproximada, NOME_COMPLETO
```
```suggestion
    return credito, idade_aproximada, nome_solicitante
```

adilsoncarvalho commented on main.py:
```diff
@@ -1,15 +1,5 @@
-    print(f"Voce possui o valor em credito de R$ {valor_credito_disponivel:.2f}, disponiveis para compras em nossa loja.")
 
-print(obter_limite())
+print("\nAperte a tecla ENTER para sair")
```
Veja a modificação que eu fiz em `modularizar`. IMHO fica mais elegante, mas o melhor seria você perguntar para o usuário.

```suggestion
solicitar_emprestimo("Rodiney Branta")
print("\nAperte a tecla ENTER para sair")
```

adilsoncarvalho reviewed (changes requested):
Melhorou bastante. Considerando os pontos que eu indiquei, na próxima eu aprovo direto 🤣

BrantaRodiney committed:
Update modularizar.py

Aceito a sugestao

Co-authored-by: Adilson Carvalho <lc.adilson@gmail.com>

marcocarvalho reviewed (commented):
Acho que com as sugestões do Adilson deu, fechou.

Depois deste PR aprovado e mergeado ao master, vale a pena você fazer um PR só para incluir testesa automatizados no seu projeto.

Para isso vc vai precisar de 
- gerenciador de dependencias
- framework de testes para python
